torchaudio/functional/functional.py

```diff
@@ -971,3 +975,34 @@ def spectral_centroid(
                            device=specgram.device).reshape((-1, 1))
     freq_dim = -2
     return (freqs * specgram).sum(dim=freq_dim) / specgram.sum(dim=freq_dim)
+
+
+def apply_codec(waveform, sample_rate, format, channels_first=True, compression=None) -> Tensor:
+    r"""
+    Applies codecs as a form of augmentation
+
+    Args:
+        waveform (Tensor): Tensor of audio of dimension (..., time)
+        sample_rate (int): Sample rate of the audio waveform
+        format (str): file format
+        channels_first (bool):
+            When True, the returned Tensor has dimension ``[channel, time]``.
+            Otherwise, the returned Tensor's dimension is ``[time, channel]``.
+        compression (float):
+            Used for formats other than WAV. This corresponds to ``-C`` option of ``sox`` command.
+                * | ``MP3``: Either bitrate (in ``kbps``) with quality factor, such as ``128.2``, or
+                  | VBR encoding with quality factor such as ``-4.2``. Default: ``-4.5``.
+                * | ``FLAC``: compression level. Whole number from ``0`` to ``8``.
+                  | ``8`` is default and highest compression.
+                * | ``OGG/VORBIS``: number from ``-1`` to ``10``; ``-1`` is the highest compression
+                  | and lowest quality. Default: ``3``.
+            See the detail at http://sox.sourceforge.net/soxformat.html.
+
+    Returns:
+        Tensor: Dimension (..., time)
```

Comment:
Similar to the above `(..., time)` is not accurate. Can you also mention that the number of frames might change for certain codecs?

Reply:
Sorry, I didn't get this comment correctly. 
Since number of frames was not used inside this method, I didn't understand how to add it in the return statement inside the docstrings. Is it part of the shape of the output tensor? is our output tensor something like (..., num_frames)?

Reply:
You can use the something similar to [the docstring of I/O function](https://github.com/pytorch/audio/blob/3651412bbcb0c4faf56bc7be7fe5e5ae2a7a63fd/torchaudio/backend/sox_io_backend.py#L155-L159). If ``channels_first=True``, it has
            ``[channel, time]`` else ``[time, channel]``.